docs/architecture/cluster-registration.md

```diff
@@ -0,0 +1,25 @@
+# Cluster registry workflow
+
+The basic registration process is that the agent first sends a registration request to the hub cluster through the bootstrap token of the hub cluster. After the hub cluster detects the registration request, the hub cluster will start to create some necessary resources for registration. The registration request will be marked as approved after these resources to be created and ready.  When the cluster registration success, the agent will periodically send its heartbeat information to the hub cluster.
```

Comment:
1. In order to make it easy to understand, add an overview to this doc at the beginning. "Registering a cluster in Clusternet is similar to joining a new node to Kubernetes."